backend/app/core/permissions.py

```
@@ -501,4 +417,58 @@ async def get_effective_permissions(user_id: int, db: AsyncSession) -> dict:
+        
+        for key in list(role_permission_cache.keys()):
+            if str(key).startswith(prefix):
+                keys_to_remove.append(('role', key))
+                
+        for key in list(department_permission_cache.keys()):
+            if str(key).startswith(prefix):
+                keys_to_remove.append(('dept', key))
+        
+        for cache_type, key in keys_to_remove:
+            if cache_type == 'user':
+                await anyio.to_thread.run_sync(lambda: user_permission_cache.pop(key, None))
```

Ruff (0.12.2)

Function definition does not bind loop variable `key`

(B023)